dpnp/backend/kernels/elementwise_functions/interpolate.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <sycl/sycl.hpp>
+#include <vector>
+
+#include "ext/common.hpp"
+#include "utils/type_utils.hpp"
+
+namespace type_utils = dpctl::tensor::type_utils;
+
+using ext::common::IsNan;
+
+namespace dpnp::kernels::interpolate
+{
+template <typename TCoord, typename TValue>
+sycl::event interpolate_impl(sycl::queue &q,
+                             const TCoord *x,
+                             const std::int64_t *idx,
+                             const TCoord *xp,
+                             const TValue *fp,
+                             const TValue *left,
+                             const TValue *right,
+                             TValue *out,
+                             const std::size_t n,
+                             const std::size_t xp_size,
+                             const std::vector<sycl::event> &depends)
+{
+    return q.submit([&](sycl::handler &h) {
+        h.depends_on(depends);
+        h.parallel_for(sycl::range<1>(n), [=](sycl::id<1> i) {
```

Comment:
It would be helpful to leave a comment why it was decided to use `sycl::id<1>`, but not `sycl::nd_item<1>` as we used in many other kernels.

Reply:
This version of kernel implementation is selected over the work-group version because it shows slightly better performance likely due to simpler execution without additional group and loop overhead

Comparison of two versions with different WorkPI on PVC

<img width="668" alt="image" src="https://github.com/user-attachments/assets/b804124c-572e-48b2-8b9f-166f2e72f262" />